### PR TITLE
Add txHash property to 'step' event object #288

### PR DESCRIPTION
**Pull request by zigen**
#288

### CODE REVIEW

**Comment by zigen** [comment]
tests failed due to this issue https://github.com/ethereumjs/ethereumjs-vm/issues/294

**Comment by holgerd77** [comment]
The tests you are referencing for failure are the VM tests, which are not executed on the Travis test runs. The tests failing are blockchain tests, so this shouldn't be related.

**Comment by holgerd77** [comment]
Ok, cool

**Comment by holgerd77** [comment]
Hi @zigen, would you regard this as ready for merge? If so can you rebase (if you haven't done before, make a local copy of your repository before) then I can review.

**Comment by zigen** [comment]
Hi @holgerd77 , this is ready for review! so I rebased and pushed.

File: lib/runCall.js [hunk 1]
```diff
@@ -18,6 +18,7 @@ const ERROR = exceptions.ERROR
  * @param opts.gasPrice {Buffer}
  * @param opts.origin {Buffer} []
  * @param opts.to {Buffer}
+ * @param opts.txHash {Buffer}
```
Comment (holgerd77) [inline lib/runCall.js]: Hmm, not sure, maybe I don't understand this fully, but why does ``txHash`` also goes here (and in ``runCode`` as well) go in the options?

Comment (zigen) [reply lib/runCall.js]: sorry for late reply and thanks for reviewing!
`runCode.js` emits `step` event here
https://github.com/ethereumjs/ethereumjs-vm/blob/master/lib/runCode.js#L128
And runBlock calls runTx -> runCall -> runCode, but `runCode` and `runCall` don't know txHash, so we need pass txHash to runCall.

**Comment by holgerd77** [comment]
Hi @zigen, thanks for reaching out today! Sorry, but I am generally not happy that we reach some values throughout the API of half of the call-stack just to print them out somewhere. Maybe this only goes together with some major general refactoring on the interplay of the different models, also have to put some more thought into that.

Will leave this PR open for now and don't merge.

**Comment by holgerd77** [comment]
PR not accepted in the current form, will close here.

**Pull request closed without merging**